src/runtime/stopwatch.ts

```diff
@@ -0,0 +1,81 @@
+import { messages } from '@cucumber/messages'
+import { stopwatch, Stopwatch, duration, Duration } from 'durations'
+
+export interface ITestRunStopwatch {
+  from: (duration: Duration) => ITestRunStopwatch
+  start: () => ITestRunStopwatch
+  stop: () => ITestRunStopwatch
+  duration: () => Duration
+  timestamp: () => messages.ITimestamp
+}
+
+export class RealTestRunStopwatch implements ITestRunStopwatch {
+  private readonly stopwatch: Stopwatch = stopwatch()
+  private base: Duration = null
+
+  from(duration: Duration): ITestRunStopwatch {
+    this.base = duration
+    return this
+  }
+
+  start(): ITestRunStopwatch {
+    this.stopwatch.start()
+    return this
+  }
+
+  stop(): ITestRunStopwatch {
+    this.stopwatch.stop()
+    return this
+  }
+
+  duration(): Duration {
+    const current = this.stopwatch.duration()
+    if (this.base !== null) {
+      return duration(this.base.nanos() + current.nanos())
+    }
+    return current
+  }
+
+  timestamp(): messages.ITimestamp {
+    const current = this.duration()
+    return {
+      nanos: current.nanos(),
```

Comment:
`nanos()` returns the whole duration in nanos whereas I think we want only the second fraction. 

Along the same line `seconds()` returns a fraction whereas I think we want a whole number.

https://github.com/joeledwards/node-durations#methods-1

Reply:
Right, duh, I’ll fix this and have a look at the timing mismatch too.

Reply:
Okay, pushed a fix for the timestamp seconds/nanos.

With the durations, when testing locally I can see the same behaviour on this branch and master - the summary formatter's total duration is on average about 67-68% of the actual elapsed time. Agree with your theory, it's the accumulation of user code without the cucumber plumbing in between. Whereas `testRunFinished.timestamp` is more because it's the actual time elapsed from when we emit `testRunStarted` (confirmed this is accurate with some logging to make sure).

On this basis should we change the summary formatter to use the same result as the message?

Reply:
Thanks for sorting the timestamps! Think we are finally there now